server/src/main/java/org/elasticsearch/cluster/routing/IndexRouting.java

```diff
@@ -293,12 +315,39 @@ public int updateShard(String id, @Nullable String routing) {
 
         @Override
         public int deleteShard(String id, @Nullable String routing) {
-            throw new IllegalArgumentException(error("delete"));
+            return routingToHash(routing);
         }
 
         @Override
         public int getShard(String id, @Nullable String routing) {
-            throw new IllegalArgumentException(error("get"));
+            return routingToHash(routing);
+        }
+
+        private int routingToHash(String routing) {
+            if (routing == null) {
+                throw new ResourceNotFoundException("routing is required because [{}] is in time series mode", indexName);
+            }
+            byte[] routingBytes;
+            try {
```

Comment:
I think it might be good to check the routing string length before trying to decode it here since we always expect 6 characters, right?

Reply:
I was just letting those cases through and complaining when we don't decode the right number of bytes. I was intentionally a bit permissive here because I figured we might want to change this in the future and only reading the first few bytes might give us some clever out for BWC.